File: CityTraveler.Domain/Filters/FilterTrips.cs
using CityTraveler.Domain.Enums;
using System;
using System.ComponentModel.DataAnnotations;

namespace CityTraveler.Domain.Entities
{
    public class FilterTrips
    {
        public DateTime TripStart { get; set; } = new DateTime();
        public string EntertaimentName { get; set; } = "";
        [DisplayFormat(ConvertEmptyStringToNull = false, NullDisplayText = "")]
        public string User { get; set; } = "";
        public double PriceMore { get; set; } = 0;
        public double PriceLess { get; set; } = double.MaxValue;
        public double AverageRatingMore { get; set; } = 0;
        public double AverageRatingLess { get; set; } = 5;
        [DisplayFormat(ConvertEmptyStringToNull = false, NullDisplayText = "")]
        public string Title { get; set; } = "";
        [DisplayFormat(ConvertEmptyStringToNull = false, NullDisplayText = "")]
        public string Description { get; set; } = "";
        public int TripStatus { get; set; }
    }
}

File: CityTraveler/ClientApp/src/app/models/filters/filterTrips.ts
import { TripStatus } from "src/app/enums/tripStatus";

export interface IFilterTrips {
    tripStart: Date,
    entertainmentName: string,
    user: string,
    priceMore: number,
    priceLess: number,
    averageRatingMore: number,
    averageRatingLess: number,
    title: string,
    description: string,
    tripStatus: TripStatus
}
